explain stream states (#39)
quicwg/reliable-stream-reset: draft-ietf-quic-reliable-stream-reset.md

frame does not prevent a QUIC stack from delivering data beyond the
specified offset to the receiving application.

## Stream States {#stream-states}

In terms of stream state transitions ({{Section 3 of RFC9000}}, the effect of
RESET_STREAM_AT frame is equivalent to that of the FIN bit. This is because both

> the RESET_STREAM_AT frame and the FIN bit serve the same role: signalling the

Reviewer: I believe this is the more common spelling in American English:
```suggestion
the RESET_STREAM_AT frame and the FIN bit serve the same role: signaling the
```

amount of data to be delivered.

> On the sending side, when a RESET_STREAM_AT frame is sent, the sending part of
> the stream enters the "Data Sent" state. Once the RESET_STREAM_AT frame and all
> stream data up to the Reliable Size are acknowledged, the sending part of the

Author: ```suggestion
On the sending side, when the first RESET_STREAM_AT frame is sent, the sending
part of the stream enters the "Data Sent" state. Once the RESET_STREAM_AT frame
and all stream data up to the smallest Reliable Size being sent are
acknowledged, the sending part of the
```

stream enters the "Data Recvd" state. Transition from "Data Sent" to "Data
Recvd" happens immediately when the application resets a stream and if all bytes
up to the specified Reliable Size have been sent and acknowledged already.
Conversely, the transition might take multiple roundtrips or require additional

> flow control credits issued by the receiver.

Reviewer: Should we mention how this works when multiple frames were sent?

On the receiving side, when a RESET_STREAM_AT frame is received, the receiving
part of the stream enters the "Size Known" state. Once all data up to the

> Reliable Size have been received, it enters the "Data Recvd" state. Similarly to

Author: ```suggestion
smallest Reliable Size have been received, it enters the "Data Recvd" state.
Similarly to
```
